verilog/tools/syntax/export_json_examples/verible_verilog_syntax.py

```diff
@@ -0,0 +1,329 @@
+        def transform(tree):
+            if tree is None:
+                return None
+            if "children" in tree:
+                children = [
+                    transform(child) or LeafNode()
+                        for child in tree["children"]
+                        if (not skip_null or child is not None)
+                ]
+                tag = tree["tag"]
+                return BranchNode(tag, children=children)
+            tag = tree["tag"]
+            start = tree["start"]
+            end = tree["end"]
+            return TokenNode(tag, start, end)
+
+        if "children" not in tree:
+            return None
+
+        children = [transform(child) for child in tree["children"]]
+        tag = tree["tag"]
+        return RootNode(tag, syntax_data=data, children=children)
+
+
+    @staticmethod
+    def _transform_tokens(tokens, data):
+        return [Token(t["tag"], t["start"], t["end"], data) for t in tokens]
+
+
+    @staticmethod
+    def _transform_errors(tokens):
+        return [Error(t["line"], t["column"], t["phase"], t.get("message", None))
+                for t in tokens]
+
+    def _parse(self, paths, input=None, options={}):
+        options = {
+            "gen_tree": True,
+            "skip_null": False,
+            "gen_tokens": False,
+            "gen_rawtokens": False,
+            **options,
+        }
+
+        args = ["-export_json"]
+        if options["gen_tree"]:
+            args.append("-printtree")
+        if options["gen_tokens"]:
+            args.append("-printtokens")
+        if options["gen_rawtokens"]:
+            args.append("-printrawtokens")
+
+        proc = subprocess.run([self.executable, *args , *paths],
+                stdout=subprocess.PIPE,
+                input=input,
+                encoding="utf-8")
+
+        json_data = json.loads(proc.stdout)
+        data = {}
+        for file_path, file_json in json_data.items():
+            file_data = SyntaxData()
+
+            if file_path == "-":
+                file_data.source_code = input
+            else:
+                with open(file_path, "rb") as f:
+                    file_data.source_code = f.read()
+
+            if "tree" in file_json:
+                file_data.tree = VeribleVerilogSyntax._transform_tree(
+                        file_json["tree"], file_data, options["skip_null"])
+
+            if "tokens" in file_json:
+                file_data.tokens = VeribleVerilogSyntax._transform_tokens(
+                        file_json["tokens"], file_data)
+
+            if "rawtokens" in file_json:
+                file_data.rawtokens = VeribleVerilogSyntax._transform_tokens(
+                        file_json["rawtokens"], file_data)
+
+            if "errors" in file_json:
+                file_data.errors = VeribleVerilogSyntax._transform_errors(file_json["errors"])
+
+            data[file_path] = file_data
+
+        return data
+
+    def parse_files(self, paths, options={}):
```

Comment:
General comment: can we add typing annotations to function args, e.g. `Iterable[str]`?

Reply:
done